src/modules/auth/auth-passport-jwt/controller/jwtAuthController.ts

```diff
@@ -0,0 +1,114 @@
+import { Request, Response } from 'express';
+import bcrypt from 'bcrypt';
+import UserModel from '../../../users/model/UserModel';
+import { blacklistToken, generateJWT } from '../../../../middleware/passport/strategies/jwt/jwt-utils';
+import jwt from 'jsonwebtoken';
+import UserDao from "../../../users/dao/UserDao";
+
+/**
+ * Register a new user.
+ *
+ * This function handles user registration. It hashes the user's password,
+ * saves the user to the database, generates a JWT token for the user,
+ * blacklists the token, and returns the token in the response.
+ *
+ * @param {Request} req - The Express request object.
+ * @param {Response} res - The Express response object.
+ */
+export const register = async (req: Request, res: Response) => {
+    const { username, email, passwordHash: hashPassword, role, firstName, lastName, address, phone } = req.body;
+    try {
+        // Hash the provided password
+        const passwordHash = await bcrypt.hash(hashPassword, 10);
+
+        // Initialize the UserDao with the UserModel
+        const userDao = new UserDao(UserModel);
+
+        // Create the user in the database
+        const userData = await userDao.createUser({
+            username,
+            email,
+            passwordHash,
+            role,
+            firstName,
+            lastName,
+            address,
+            phone
+        });
+
+        // Generate a JWT token for the newly created user
+        const token = generateJWT(userData);
+        const expiryDate = new Date(Date.now() + 3600 * 1000); // 1 hour
+
+        // Blacklist the token (optional step depending on your use case)
+        await blacklistToken(token, expiryDate);
```

Comment:
why is this called blacklist? is it convention of some sort?

Reply:
When a user logs out, you want to ensure that the token they were using is no longer valid. This helps prevent misuse if the token gets intercepted or if the user tries to use the same token after logging out.

Reply:
but we are saving active tokens there, right?

Reply:
no no need save active tokens for current stage if we don't have special requirement